File: php/tests/Unit/Domain/Customer/InvoiceTest.php
<?php

namespace Tests\Katalogs\BookInvoicing\Unit\Domain\Customer;

use Katalogs\BookInvoicing\Domain\Books\Author;
use Katalogs\BookInvoicing\Domain\Books\Genre;
use Katalogs\BookInvoicing\Domain\Books\Novel;
use Katalogs\BookInvoicing\Domain\Country\Country;
use Katalogs\BookInvoicing\Domain\Country\Currency;
use Katalogs\BookInvoicing\Domain\Country\Language;
use Katalogs\BookInvoicing\Domain\Purchase\Invoice;
use Katalogs\BookInvoicing\Domain\Purchase\PurchasedBook;
use PHPUnit\Framework\Assert;
use PHPUnit\Framework\TestCase;

class InvoiceTest extends TestCase
{
    /**
     * @test
     * @return void
     */
    public function should_apply_tax_rules_when_computing_total_amount(): void
    {
        $book = new Novel(
            "Beautiful novel", 50,
            new Author('An author', new Country('USA', Currency::Dollar, Language::English)),
            Language::English, Genre::Mystery
        );

        $purchase = new PurchasedBook($book, 1);

        $invoice = new Invoice(
            'John Doe',
            new Country('USA', Currency::Dollar, Language::English)
        );

        $invoice->addPurchase($purchase);

        Assert::assertEquals(56.35, $invoice->computeTotalAmount());


    }
}

File: go/test/invoice_test.go
package test

import (
	. "book_invoicing/domain/book"
	. "book_invoicing/domain/country"
	. "book_invoicing/purchase"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestInvoice_ShouldApplyTaxRules_WhenComputingTotalAmount(t *testing.T) {
	var book = NewNovel(
		"Beautiful novel",
		50,
		Author{
			Name: "An author",
			Nationality: Country{
				Name:     "USA",
				Currency: Dollar,
				Language: English,
			},
		},
		English,
		map[int64]Genre{1: Mystery},
	)
	purchase := NewPurchasedBook(book, 1)

	invoice := NewInvoice("John Doe", Country{
		Name:     "USA",
		Currency: Dollar,
		Language: English,
	})

	invoice.AddPurchasedBook(purchase)

	assert.InDelta(t, 57.50, invoice.ComputeTotalAmount(), 0.001)
}
